app/src/main/cpp/skyline/gpu.cpp

```diff
@@ -332,30 +333,27 @@ namespace skyline::gpu {
     }
 
     static PFN_vkGetInstanceProcAddr LoadVulkanDriver(const DeviceState &state) {
-        // Try turnip first, if not then fallback to regular with file redirect then plain dlopen
-        auto libvulkanHandle{adrenotools_open_libvulkan(
-            RTLD_NOW,
-            ADRENOTOOLS_DRIVER_CUSTOM,
-            nullptr, // We require Android 10 so don't need to supply
-            state.os->nativeLibraryPath.c_str(),
-            (state.os->privateAppFilesPath + "gpu/turnip/").c_str(),
-            "libvulkan_freedreno.so",
-            nullptr
-        )};
-        if (!libvulkanHandle) {
+        void *libvulkanHandle{};
+
+        // If the user has selected a custom driver, try to load it
+        if (!(*state.settings->gpuDriver).empty()) {
             libvulkanHandle = adrenotools_open_libvulkan(
                 RTLD_NOW,
-                ADRENOTOOLS_DRIVER_FILE_REDIRECT,
+                ADRENOTOOLS_DRIVER_FILE_REDIRECT | ADRENOTOOLS_DRIVER_CUSTOM,
                 nullptr, // We require Android 10 so don't need to supply
                 state.os->nativeLibraryPath.c_str(),
-                nullptr,
-                nullptr,
+                (state.os->privateAppFilesPath + "gpu_drivers/" + *state.settings->gpuDriver + "/").c_str(),
+                (*state.settings->gpuDriverLibraryName).c_str(),
                 (state.os->publicAppFilesPath + "gpu/vk_file_redirect/").c_str()
             );
+
             if (!libvulkanHandle)
-                libvulkanHandle = dlopen("libvulkan.so", RTLD_NOW);
+                Logger::Warn("Failed to load custom Vulkan driver {}/{}: {}", *state.settings->gpuDriver, *state.settings->gpuDriverLibraryName, dlerror());
```

Comment:
dlerror can return null, this might be fine but worth checking, otherwise lgtm

Reply:
Oh that's why it was crashing, thanks